library/std/src/sys/unix/ext/ucred.rs

```diff
@@ -0,0 +1,110 @@
+//! Unix peer credentials.
+
+// NOTE: Code in this file is heavily based on work done in PR 13 from the tokio-uds repository on
+//       GitHub.
+//
+//       For reference, the link is here: https://github.com/tokio-rs/tokio-uds/pull/13
+//       Credit to Martin Habovštiak (GitHub username Kixunil) and contributors for this work.
+
+use libc::{gid_t, uid_t};
+
+/// Credentials for a UNIX process for credentials passing.
+#[unstable(feature = "peer_credentials_unix_socket", issue = "42839", reason = "unstable")]
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub struct UCred {
+    pub uid: uid_t,
+    pub gid: gid_t,
+}
```

Comment:
Out of curiosity - why not return the process ID too, since that's returned by the syscall?

Reply:
PID is only available on Linux, it's not available on BSD.

Reply:
Is it possible to include it (conditionally) on Linux?

Reply:
Technically yes. It'd require change of the API due to coding standard in `std` (mainly using extension traits for additional platforms, IDK why such standard exists).

Reply:
Ah, fair enough, I was hoping for a small, clean change :) nevermind then

Reply:
Could we make the PID an `Option<pid_t>`? A quick search reveals that it is in fact supported on quite a few BSD platforms (e.g. OpenBSD).

Reply:
Done -- however, my Google-fu is failing me here. I can't find anything about retrieving the PID from a domain socket on FreeBSD. The closest I can find is `getpeereid` which accepts a UID and GID only. So at the moment, the optional PID is set to `None` for BSD, and `Some(pid)` on Linux.

Reply:
It seems to have been added relatively recently: https://svnweb.freebsd.org/base?view=revision&revision=348419